core/src/utils.rs

```diff
@@ -112,13 +112,13 @@ pub fn confirm(msg: &str) -> Result<bool> {
     }
 }
 
-#[derive(Serialize)]
+#[derive(Serialize, Deserialize)]
 pub struct APIError {
     pub code: String,
     pub message: String,
 }
 
-#[derive(Serialize)]
+#[derive(Serialize, Deserialize)]
 pub struct APIResponse {
```

Comment:
I didn't know we already had that, i introduced something similar in my integration tests.

Reply:
yeah, it was stupid of me. Will refactor to use it (unless you want to do it in this PR)

Reply:
Def not stupid, it wasn't designed for that. now that we have rust-to-rust it starts making sense